src/python/pants/core/util_rules/environments_test.py

```diff
@@ -215,32 +227,133 @@ def get_name(v: str) -> EnvironmentName:
     assert get_name(LOCAL_ENVIRONMENT_MATCHER).val is None
     # Else, error for unrecognized names.
     with engine_error(UnrecognizedEnvironmentError):
-        get_name("hardcoded")
+        get_name("local")
 
     env_names_arg = (
         "--environments-preview-names={"
         + "'local': '//:local', "
-        + "'hardcoded': '//:hardcoded', "
+        + "'local-fallback': '//:local-fallback', "
         + "'docker': '//:docker', "
         + "'remote-no-fallback': '//:remote-no-fallback', "
         + "'remote-fallback': '//:remote-fallback',"
         "'remote-bad-fallback': '//:remote-bad-fallback'}"
     )
     rule_runner.set_options([env_names_arg, "--remote-execution"])
     assert get_name(LOCAL_ENVIRONMENT_MATCHER).val == "local"
-    for name in ("hardcoded", "docker", "remote-no-fallback", "remote-fallback"):
+    for name in ("local", "docker", "remote-no-fallback", "remote-fallback"):
         assert get_name(name).val == name
     with engine_error(UnrecognizedEnvironmentError):
         get_name("fake")
 
+    assert get_name("local-fallback").val == "local"
+
     rule_runner.set_options([env_names_arg, "--no-remote-execution"])
     assert get_name("remote-fallback").val == "docker"
-    with engine_error(RemoteExecutionDisabledError):
+    with engine_error(NoFallbackEnvironmentError):
         get_name("remote-no-fallback")
     with engine_error(UnrecognizedEnvironmentError):
         get_name("remote-bad-fallback")
 
 
+def test_resolve_environment_name_local_and_docker_fallbacks(monkeypatch) -> None:
+    # We can't monkeypatch the Platform with RuleRunner, so instead use run_run_with_mocks.
```

Comment:
Hm... you _might_ be able to do it by mocking the return value of `Platform.create_for_localhost`, as that is what will drive the selection here? But either way: thanks for the thorough test.